File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        List<Pair<Integer, Integer>> pairs = new ArrayList<>();
        
        for (int i = 0; i < nums.length; i++) {
            pairs.add(new Pair<>(nums[i], i));
        }
        
        pairs.sort(Comparator.comparingInt(p -> p.getKey()));
        
        int i = 0;
        int j = pairs.size() - 1;
        
        while (i < j) {
            int sum = pairs.get(i).getKey() + pairs.get(j).getKey();
            if (sum == target) {
                return new int[]{pairs.get(i).getValue(), pairs.get(j).getValue()};
            } else if (sum < target) {
                i++;
            } else {
                j--;
            }
        }
        
        return new int[0];
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int, int>> v;
        vector<int> ans;
        for (int i = 0; i < nums.size(); i++) {
            v.push_back({nums[i], i});
        }
        sort(v.begin(), v.end());
        int i = 0;
        int j = nums.size() - 1;
        while (i < j) {
            int x = (v[i].first + v[j].first);
            if (x == target) {
                ans.push_back(v[i].second);
                ans.push_back(v[j].second);
                return ans;
            } else if (x > target) {
                j--;
            } else {
                i++;
            }
        }
        return ans;
    }
};